lib/Unicode/Char/General.hs

```diff
@@ -75,29 +91,391 @@ where
+generalCategoryAbbr ConnectorPunctuation = "Pc"
+generalCategoryAbbr DashPunctuation      = "Pd"
+generalCategoryAbbr OpenPunctuation      = "Ps"
+generalCategoryAbbr ClosePunctuation     = "Pe"
+generalCategoryAbbr InitialQuote         = "Pi"
+generalCategoryAbbr FinalQuote           = "Pf"
+generalCategoryAbbr OtherPunctuation     = "Po"
+generalCategoryAbbr MathSymbol           = "Sm"
+generalCategoryAbbr CurrencySymbol       = "Sc"
+generalCategoryAbbr ModifierSymbol       = "Sk"
+generalCategoryAbbr OtherSymbol          = "So"
+generalCategoryAbbr Space                = "Zs"
+generalCategoryAbbr LineSeparator        = "Zl"
+generalCategoryAbbr ParagraphSeparator   = "Zp"
+generalCategoryAbbr Control              = "Cc"
+generalCategoryAbbr Format               = "Cf"
+generalCategoryAbbr Surrogate            = "Cs"
+generalCategoryAbbr PrivateUse           = "Co"
+generalCategoryAbbr NotAssigned          = "Cn"
+
+{-| The Unicode general category of the character.
+
+This property is defined in the column 2 of the @UnicodeData@ table.
+
+This relies on the 'Enum' instance of 'GeneralCategory', which must remain in the
+same order as the categories are presented in the Unicode standard.
+
+prop> show (generalCategory c) == show (Data.Char.generalCategory c)
+-}
+{-# INLINE generalCategory #-}
+generalCategory :: Char -> GeneralCategory
+generalCategory = toEnum . UC.generalCategory
+
+{-| Selects alphabetic Unicode characters (lower-case, upper-case and title-case letters, plus letters of caseless scripts and modifiers letters).
+
+This function returns 'True' if its argument has one of the
+following 'GeneralCategory's, or 'False' otherwise:
+
+* 'UppercaseLetter'
+* 'LowercaseLetter'
+* 'TitlecaseLetter'
+* 'ModifierLetter'
+* 'OtherLetter'
+
+Alias of 'isLetter'.
+
+__Note:__ this function is /not/ equivalent to 'isAlphabetic'.
+See the description of 'isAlphabetic' for further details.
+
+prop> isAlpha c == Data.Char.isAlpha c
+-}
+{-# INLINE isAlpha #-}
+isAlpha :: Char -> Bool
+isAlpha = isLetter
+
+{-| Returns 'True' for alphabetic Unicode characters (lower-case, upper-case
+and title-case letters, plus letters of caseless scripts and modifiers
+letters).
+
+__Note:__ this function is /not/ equivalent to 'isAlpha'/'isLetter':
+
+* 'isAlpha' matches the following general categories:
+
+    * 'UppercaseLetter' (@Lu@)
+    * 'LowercaseLetter' (@Ll@)
+    * 'TitlecaseLetter' (@Lt@)
+    * 'ModifierLetter' (@Lm@)
+    * 'OtherLetter' (@Lo@)
+
+* whereas 'isAlphabetic' matches:
+
+    * @Uppercase@ [property](https://www.unicode.org/reports/tr44/#Uppercase)
+    * @Lowercase@ [property](https://www.unicode.org/reports/tr44/#Lowercase)
+    * 'TitlecaseLetter' (@Lt@)
+    * 'ModifierLetter' (@Lm@)
+    * 'OtherLetter' (@Lo@)
+    * 'LetterNumber' (@Nl@)
+    * @Other_Alphabetic@ [property](https://www.unicode.org/reports/tr44/#Other_Alphabetic)
+-}
+{-# INLINE isAlphabetic #-}
+isAlphabetic :: Char -> Bool
+isAlphabetic = P.isAlphabetic
```

Comment:
Fine with me, no strong opinion.

Reply:
My reasoning on this is:
- Predicates are named from their property in Unicode. They may have no equivalent in `Data.Char`.
- `base` predicates are kept unchanged.
- Having `Unicode.Char` a drop-in replacement of `Data.Char` make it possible to ensure a specific version of Unicode rather than depend on GHC’s Unicode version.

I agree that having predicates that differ only for a few characters is confusing. I would rather use the ones from Unicode when possible. Therefore I propose the following:
- Move `isAlpha`, `isLetter` and `isSpace` to a new module `Unicode.Char.General.Compat`.
- Re-export `Unicode.Char.General.Compat` in `Unicode.Char`.

Reply:
Sounds good. I am fine with it even the way it is now, you can decide whichever way you like. We can mark them `Compat` APIs in the haddock and suggest the better/recommended alternative. Anyway, the benchmarks look favorable for the new APIs, so that's another reason to use those.

Reply:
I moved the functions to a compatiblity module.